test/services/videoconference/index.test.js

```diff
@@ -0,0 +1,206 @@
+		const { id: schoolId } = school;
+		const courseTeacher = await testObjects.createTestUser({
+			roles: ['teacher'],
+			schoolId,
+			firstName: 'teacher',
+		});
+		const someUser = await testObjects.createTestUser({
+			roles: ['teacher'],
+			schoolId,
+			firstName: 'a user',
+		});
+		const coursesubstitutionTeacher = await testObjects.createTestUser({
+			roles: ['teacher'],
+			schoolId,
+			firstName: 'substitutionteacher',
+		});
+		const courseStudent = await testObjects.createTestUser({
+			roles: ['student'],
+			schoolId,
+			firstName: 'student',
+		});
+		const teacherRequestAuthentication = await generateRequestParamsFromUser(courseTeacher);
+		const someUserAuth = await generateRequestParamsFromUser(someUser);
+		const substitutionTeacherRequestAuthentication = await generateRequestParamsFromUser(coursesubstitutionTeacher);
+		const studentRequestAuthentication = await generateRequestParamsFromUser(courseStudent);
+		const course = await testObjects.createTestCourse({
+			name: 'test videoconference course',
+			userIds: [courseStudent.id],
+			teacherIds: [courseTeacher.id],
+			substitutionIds: [coursesubstitutionTeacher.id],
+		});
+		const serviceParams = {
+			scopeId: course.id,
+			scopeName: 'course',
+		};
+		testData = {
+			school,
+			courseTeacher,
+			courseStudent,
+			someUser,
+			someUserAuth,
+			teacherRequestAuthentication,
+			substitutionTeacherRequestAuthentication,
+			studentRequestAuthentication,
+			course,
+			serviceParams,
+		};
+	});
+
+	it('fails with school feature disabled, enables school feature for further tests', async () => {
+		expect(() => createService
+			.create(testData.reateOptions, testData.teacherRequestAuthentication), 'feature probably enabled in school')
+			.to.throw;
+		testData.school.features.push(VIDEOCONFERENCE);
+		await testData.school.save();
+		const response = await createService.create(testData.serviceParams, testData.teacherRequestAuthentication);
+		expect(response).to.be.ok;
+		expect(response.status).to.be.equal('SUCCESS');
+		expect(response.url).to.be.not.empty;
+	});
+
+	it('test creation with start permission works multiple times [slow]', async () => {
+		let successfulRuns = 0;
+		const authenticated = [];
+		for (let i = 0; i < 20; i += 1) {
+			const response = await createService.create(testData.serviceParams, testData.teacherRequestAuthentication);
+			expect(response).to.be.ok;
+			expect(response.status).to.be.equal('SUCCESS');
+			expect(response.url).to.be.not.empty;
+			authenticated.push(rp(response.url));
+			successfulRuns += 1;
+		}
+		expect(successfulRuns).to.be.equal(20);
+		// expect all requests finish successfully
+		return Promise.all(authenticated);
+	});
+
+	it('test creation with join permission (student in courses) fails', () => {
+		expect(() => createService
+			.create(testData.serviceParams, testData.studentRequestAuthentication),
+			'students should not have the permission to start a videoconference in courses')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 2 tabs but found 3.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4819756)